### PR TITLE
Add android support of x86 and x86_64 emulators

### PR DESCRIPTION
**Pull request by tadovas**
Title explains it all

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -42,37 +40,29 @@ jobs:
...
-    # Official release (on tags only)
-    - stage: release
-      script:
+    - script:
       - go get golang.org/x/mobile/cmd/gomobile
-      - mkdir -p tmp artifacts
-      - ./gomobile_ios.sh -o tmp/Openvpn3.framework
-      - tar -zcvf artifacts/Openvpn3.framework.tar.gz tmp/Openvpn3.framework
+      #builds desktop and mobile versions - puts into build dir
+      - ./check-all.sh
```
Comment (soffokl) [inline .travis.yml]: we are doing `./check-all.sh` both in `test` and `release` stages. Looks like we need to get rid of one of them.

File: scripts/x-compile-android-amd64.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+echo BUILD DEPS
+pushd $DEP_DIR
+rm -rf asio* lz4* mbedtls* #lzo* boost* minicrypto openssl* polarssl* snappy*
+echo "******* ASIO"
+$O3/core/deps/asio/build-asio
+echo "******* MBEDTLS"
+TARGETS=android-amd64 $O3/core/scripts/android/build-mbedtls
+echo "******* LZ4"
+TARGETS=android-amd64 $O3/core/scripts/android/build-lz4
```
Comment (soffokl) [inline scripts/x-compile-android-amd64.sh]: Are we replacing `TARGETS` that was set 2 lines above?

File: scripts/x-compile-android-x86.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
```
Comment (soffokl) [inline scripts/x-compile-android-x86.sh]: This looks close to `scripts/x-compile-android-amd64.sh` file. Can we do a common script with architecture argument?

File: xgo-check.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```
Comment (soffokl) [inline xgo-check.sh]: Why it names `xgo-check.sh`, what does it check?

File: scripts/x-compile-android-amd64.sh (hunk 2 above)
Comment (tadovas) [reply scripts/x-compile-android-amd64.sh]: Nope. But keep in mind that TARGETS variable is expected inside those scripts. It's not possible to set once and pass to both scripts without actually doing export.

File: .travis.yml (hunk 1 above)
Comment (tadovas) [reply .travis.yml]: On first time we are doing build to see if it at least builds - kind of a check. On release stage we take actual build artifacts and attach them to released tag. Actually I am not sure if they are of any use, as the whole point of this repo is to be go gettable package, not a standalone product. Possibly gonna get rid of release section.

File: scripts/x-compile-android-x86.sh (hunk 3 above)
Comment (tadovas) [reply scripts/x-compile-android-x86.sh]: Yea, probably. But maybe next time - those scripts are more like - run and forget, maybe we will get rid of them completely some day.

File: xgo-check.sh (hunk 4 above)
Comment (tadovas) [reply xgo-check.sh]: It checks if package being tested actually can be build

**Pull request merged**